app/services/groups/create_service.rb

```diff
@@ -3,20 +3,31 @@
 module Groups
   # Service used to Create Groups
   class CreateService < BaseService
+    GroupCreateError = Class.new(StandardError)
+    attr_accessor :group
+
     def initialize(user = nil, params = {})
       super(user, params)
+      @group = Group.new(params.merge(owner: current_user))
     end
 
-    def execute
-      @group = Group.new(params.merge(owner: current_user))
-      @group.save
+    def execute # rubocop:disable Metrics/AbcSize
+      unless allowed_to_modify_group?(group)
+        raise GroupCreateError, I18n.t('services.groups.create.no_permission',
+                                       namespace_type: group.type.downcase)
+      end
+
+      group.save
 
-      Members::CreateService.new(current_user, @group, {
+      Members::CreateService.new(current_user, group, {
```

Comment:
We only need to create a Membership if the `group.parent` is nil, as they would already have to be an Owner in the hierarchy if they could create a subgroup. This could also be done in a separate PR.

Reply:
Updated in 81beea8257c5b723281c071ae46caeb8cbe85653